FSW/src/util/memUtil.cpp

```diff
@@ -0,0 +1,250 @@
+ *  sample - data to be stored (type must match dataBuffer type)
+ *  index - index to store new data at (pass-by-reference)
+ *  
+ * Outputs:
+ *  none
+ */
+void updateBuffer(float sample, int &index){
+    
+    // check that index is valid
+    if (0 <= index && index < BUFFERSIZE){
+        dataBuffer[index] = sample;
+        index++;
+    
+    } else {
+        Serial.print("WARNING: invalid dataBuffer index ");
+        Serial.print("(Science Memory Handling Module)\n");
+    }
+
+    if (index == BUFFERSIZE){
+        // reset index if we have reached end of buffer
+        index = 0; 
+    }  
+}
+
+/* - - - - - - saveBuffer - - - - - - *
+ * Belongs to Science Memory Handling Module
+ *  
+ * Usage:
+ *  saves the buffer array passed in to a file on the flash module in csv format
+ *  appends timestamp to the end of the file
+ * 
+ * Inputs:
+ *  dataBuffer - pointer to first element of data buffer
+ *  index - index to store new data at (pass-by-reference)
+ *  
+ * Outputs:
+ *  file creation status
+ */
+bool saveBuffer(int &index){
+    
+    // create array to hold data in time ascending order
+    float timeSortArray[BUFFERSIZE];
+    int j = 0; // iterator for sorted array index
+
+    // reorder array so that it is ascending in time
+    for (int i = index; i < BUFFERSIZE; i++){
+        timeSortArray[j] = dataBuffer[i];
+        j++;
+    } 
+
+    // loop back to top of array and store remaining values
+    for (int i = 0; i < index; i++){
+        timeSortArray[j] = dataBuffer[i];
+        j++;
+    }
+
+    /* send sorted array to file on flash memory along with timestamp 
+     * see SerialFlash docs for info on these functions
+     * https://github.com/PaulStoffregen/SerialFlash/blob/master/README.md 
+     */
+
+    // check if file exists
+    int fileIdx; // iterator for loop checking file existence
+    bool fileFlag = true;
+    char filename[] = "scienceFile0.csv";   // null-terminated char array
+    int fileIdxOffset = 11;                 // index of file number
+
+    while (fileFlag){
+        if (fileIdx < MAXFILES){ // prevent infinite loop
+            filename[fileIdxOffset] += fileIdx; // iterate up from zero
+            fileFlag = SerialFlash.exists(filename);
+            fileIdx++;
+        }
+    }
+
+    // create new file (non-erasable, delete file after downlink)
+    bool status = true; // track file creation/writing status
+    status = SerialFlash.create(filename, FILESIZE);
+
+    if (status){
+        Serial.print("Found file ");
+        Serial.print(filename);
+        Serial.print(" on flash chip");
+    }
+
+    // write buffer to this new file
+    SerialFlashFile file;
+    file = SerialFlash.open(filename);
+    status = file.write(dataBuffer, BUFFERSIZE); //write dataBuffer
+    
+    // TODO: may need to seek position to end of file before writing again
+    //file.seek(BUFFERSIZE);  
+
+    //TODO: decide on timestamp format and append it to file
```

Comment:
I think Julian time would be good here, otherwise I want to say ISO 8601 is the standard.